File: introductionToAlgorithmCLRS/chp6/InChaptAlgo/master.c
#include <stdio.h>

#include "HeapSort.h"


void main() {
    printf("Enter the size of array : ");
    int size;
    scanf("%d", &size);

    int array[size];
    printf("Enter the elements of the array : ");

    for (int i = 0 ; i < size; i++) {
        scanf("%d", &array[i]);
    }
    printf("Your given array is : [ ");
    for (int i = 0; i < size; i++) {
        printf("%d ", array[i]);
    }
    printf("]\n");

    Build(array, size);

    printf("Build Max Heap is : [ ");
    for (int i = 0; i < size; i++) {
        printf("%d ", array[i]);
    }
    printf("]\n");

    Sort(array, size);
    printf("Your sorted array is : [ ");
    for (int i = 0; i < size; i++) {
        printf("%d ", array[i]);
    }
    printf("]\n");
}

File: introductionToAlgorithmCLRS/chp6/InChaptAlgo/master.java
import java.util.Scanner; 
import java.util.Arrays; 

public class master extends HeapSort{

    public static void main(String[] args) {
        Scanner element = new Scanner(System.in);
        System.out.print("Enter the size of array : ");
        int size = element.nextInt();

        int array[] = new int[size];
        System.out.print("Enter the elements of the array : ");

        for (int i = 0 ; i < size; i++) {
            array[i] = element.nextInt();
        }
        element.close();
        System.out.println("Your given array is : "+Arrays.toString(array));
        Build(array, size);
        System.out.println("Build Max Heap is : " +Arrays.toString(array));
        Sort(array, size);
        System.out.println("Your sorted array is : "+Arrays.toString(array));

    }
}